File: typescript-connector/src/models/AsPlannedLocalConsumptionFlow.ts
// Class from DFC Business Ontology: #AsPlannedLocalConsumptionFlow
import { SemanticObject } from "../core/SemanticObject.js";
import { ConsumptionFlow, type ConsumptionFlowParams } from "./ConsumptionFlow.js";

export interface AsPlannedLocalConsumptionFlowParams extends ConsumptionFlowParams {}
export class AsPlannedLocalConsumptionFlow extends ConsumptionFlow {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:AsPlannedLocalConsumptionFlow";
  }



  constructor(
    semanticId: string,
    params?: AsPlannedLocalConsumptionFlowParams,
  ) {
    super(semanticId, params);
    this.semanticType = AsPlannedLocalConsumptionFlow.SEMANTIC_TYPE;
  }
  static {
    SemanticObject.typeRegistry.set(AsPlannedLocalConsumptionFlow.SEMANTIC_TYPE, AsPlannedLocalConsumptionFlow);
  }
}

File: ruby-gem/lib/models/as_planned_local_consumption_flow.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #AsPlannedLocalConsumptionFlow
require_relative '../core/semantic_object'
require_relative 'consumption_flow'


module DfcLinkmlConnector
  module Models
    class AsPlannedLocalConsumptionFlow < ConsumptionFlow
      SEMANTIC_TYPE = "dfc-b:AsPlannedLocalConsumptionFlow".freeze

      # @param semanticId [String]
      # @param quantity: nil, consumes: nil, inputOf: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil
      def initialize(semanticId, quantity: nil, consumes: nil, inputOf: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil)
        super(semanticId, quantity: quantity, consumes: consumes, inputOf: inputOf, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)

        self.semanticType = "dfc-b:AsPlannedLocalConsumptionFlow"

      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
